packages/server/tests/acceptance/htsPrecompile.spec.ts

```diff
@@ -34,60 +34,147 @@ describe('HTS Precompile Acceptance Tests', async function() {
+    const txCO = await baseHTSContractOwner.associateTokenPublic(baseHTSContractAddress, NftHTSTokenContractAddress, { gasLimit: 10000000 });
+    expect((await txCO.wait()).events.filter(e => e.event === 'ResponseCode')[0].args.responseCode).to.equal(22);
+
+    const baseHTSContractReceiverWalletFirst = new ethers.Contract(baseHTSContractAddress, BaseHTSJson.abi, accounts[1].wallet);
+    const txRWF = await baseHTSContractReceiverWalletFirst.associateTokenPublic(accounts[1].wallet.address, NftHTSTokenContractAddress, { gasLimit: 10000000 });
+    expect((await txRWF.wait()).events.filter(e => e.event === 'ResponseCode')[0].args.responseCode).to.equal(22);
+
+    const baseHTSContractReceiverWalletSecond = new ethers.Contract(baseHTSContractAddress, BaseHTSJson.abi, accounts[2].wallet);
+    const txRWS = await baseHTSContractReceiverWalletSecond.associateTokenPublic(accounts[2].wallet.address, NftHTSTokenContractAddress, { gasLimit: 10000000 });
+    expect((await txRWS.wait()).events.filter(e => e.event === 'ResponseCode')[0].args.responseCode).to.equal(22);
   });
 
   it('should check initial balances', async function() {
+    const HTSTokenContract = new ethers.Contract(HTSTokenContractAddress, ERC20MockJson.abi, accounts[0].wallet);
     expect(await HTSTokenContract.balanceOf(accounts[0].wallet.address)).to.equal(1000);
     expect(await HTSTokenContract.balanceOf(accounts[1].wallet.address)).to.equal(0);
     expect(await HTSTokenContract.balanceOf(accounts[2].wallet.address)).to.equal(0);
   });
 
   it('should be able to transfer hts tokens between accounts', async function() {
     const amount = 10;
-    const balanceBefore = await HTSTokenContract.balanceOf(accounts[1].wallet.address);
-
-    await baseHTSContract.transferTokenTo(accounts[1].wallet.address, HTSTokenContract.address, amount);
+    const HTSTokenContract = new ethers.Contract(HTSTokenContractAddress, ERC20MockJson.abi, accounts[0].wallet);
+    const baseHTSContract = new ethers.Contract(baseHTSContractAddress, BaseHTSJson.abi, accounts[0].wallet);
 
+    const balanceBefore = await HTSTokenContract.balanceOf(accounts[1].wallet.address);
+    await baseHTSContract.transferTokenPublic(accounts[1].wallet.address, HTSTokenContractAddress, amount);
     const balanceAfter = await HTSTokenContract.balanceOf(accounts[1].wallet.address);
+
     expect(balanceBefore + amount).to.equal(balanceAfter);
   });
+
+  it('should be able to check allowance', async function() {
+    const baseHTSContract = new ethers.Contract(baseHTSContractAddress, BaseHTSJson.abi, accounts[0].wallet);
+    const tx = await baseHTSContract.allowancePublic(HTSTokenContractAddress, baseHTSContractAddress, accounts[2].wallet.address);
+    const { responseCode } = (await tx.wait()).events.filter(e => e.event === 'ResponseCode')[0].args;
+
+    expect(responseCode).to.equal(TX_SUCCESS_CODE);
```

Comment:
Move this check into the next test `should be able to approve anyone to spend tokens` and you should be able to remove this test.
For unit tests it's good to have it broken out but for E2E acceptance you can't do anything without success and it calls cost HBAR so no need for the breakout